pkgs/tools/security/gnupg/21.nix

```diff
@@ -23,9 +23,14 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo
-    readline libusb gnutls adns openldap zlib bzip2
-  ];
+    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo ]
+      ++ optional (adns     != null) adns
```

Comment:
@leenaars while this pattern is used, it's unclear to me what it is supposed to accomplish in the simple case where the only decision being made is whether the value is added to the build inputs. As @peti says, either you have the package or `null`, and `null`s are simply ignored; checking whether the value is `null` before adding it appears redundant.